gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +473,84 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  @VisibleForTesting
+  boolean isGoogleS2AEnabled() {
+    String S2AEnv = envProvider.getenv(S2A_ENV_ENABLE_USE_S2A);
+    return Boolean.parseBoolean(S2AEnv);
+  }
+
+  @VisibleForTesting
+  boolean shouldUseS2A() {
+    // If EXPERIMENTAL_GOOGLE_API_USE_S2A is not set to true, skip S2A.
+    if (!isGoogleS2AEnabled()) {
+      return false;
+    }
+
+    // If {@code mtlsEndpoint} is not set, or {@code endpointOverride} is specified, skip S2A.
+    if (mtlsEndpoint.isEmpty() || !endpointOverride.isEmpty()) {
```

Comment:
hmm, I see `endpointOverride` is needed within transport channel. Is providing this override option a requirement for this feature or AIP?

nit: Do you mind to separate these 2 scenarios in single condition if statements for a bit more clarity?